activestorage/app/models/active_storage/preview.rb

```diff
@@ -53,7 +53,11 @@ def processed
 
   # Returns the blob's attached preview image.
   def image
-    blob.preview_image
+    if variant?
+      variant.processed.image if processed?
```

Comment:
i delegate to the variant `#image` method, which can either be a AS::Blob ( for tracked variants) or `AS::Variant` ruby object, non AR, instance.